edb/edgeql/compiler/stmt.py

```diff
@@ -216,6 +217,37 @@ def compile_ForQuery(
             node.attach_subtree(view_scope_info.path_scope,
                                 context=iterator.context)
 
+    return iterator_stmt
+
+
+@dispatch.compile.register(qlast.ForQuery)
+def compile_ForQuery(
+        qlstmt: qlast.ForQuery, *, ctx: context.ContextLevel) -> irast.Set:
+    if rewritten := try_desugar(qlstmt, ctx=ctx):
+        return rewritten
+
+    with ctx.subquery() as sctx:
+        stmt = irast.SelectStmt(context=qlstmt.context)
+        init_stmt(stmt, qlstmt, ctx=sctx, parent_ctx=ctx)
+
+        if len(qlstmt.iterator_bindings) != 1 and not ctx.env.options.devmode:
+            raise errors.UnsupportedFeatureError(
+                "Multiple 'FOR' iterators is an internal testing feature",
+                context=qlstmt.context,
+            )
+
+        iterator_stmts = []
+        for binding in qlstmt.iterator_bindings:
+            # XXX: can one iterator reference an earlier one?
+            iterator_stmts.append(
+                _compile_for_binding(qlstmt, binding, ctx=sctx))
```

Comment:
This would be equivalent to `LATERAL` in SQL.

I'd say no, let's keep it simple for now.